package/kedro_viz/data_access/managers.py

```diff
@@ -149,15 +149,18 @@ def add_pipeline(self, registered_pipeline_id: str, pipeline: KedroPipeline):
             pipeline: The Kedro pipeline instance to convert to graph models
                 and add to relevant repositories representing the graph.
         """
-        modular_pipelines = self.modular_pipelines[registered_pipeline_id]
         self.registered_pipelines.add_pipeline(registered_pipeline_id)
+        modular_pipelines_repo_obj = self.modular_pipelines[registered_pipeline_id]
+        modular_pipelines_repo_obj.populate_tree(pipeline)
+
         free_inputs = pipeline.inputs()
 
         for node in pipeline.nodes:
-            task_node = self.add_node(registered_pipeline_id, node)
-            self.registered_pipelines.add_node(registered_pipeline_id, task_node.id)
+            task_node = self.add_node(
```

Comment:
since there are two `add_node` functions; it does affect readability. Maybe we add comments before the add_node functions explaining what happens?